app/controllers/secrets_controller.rb

```diff
@@ -66,8 +66,20 @@ def batch
       unless (secret = variable.last_secret)
         raise Exceptions::RecordNotFound, variable.resource_id
       end
-      
-      result[variable.resource_id] = secret.value
+
+      mime_type = variable.annotation('conjur/mime_type') || nil
+
+      if mime_type == 'application/octet-stream'
+        unless result.key?('application/octet-stream')
+          result['application/octet-stream'] = { 'encoding': 'base64' }
```

Comment:
I think base64 encoding is robust enough to address the current issue and potentially moving forward. With that in mind, perhaps we can always base64 encode values and always specify the mimeType. This throws out any space savings etc., but I don't think that's altogether necessary and might really just be premature optimisation. The response payload might look something like this.

By avoiding sending raw values we don't ever have to know or reason about what is inside.
```
{
  "id/of/secret": {
    "base64value": "",
    "mimeType": ""
  },
  ...
}
```

Reply:
I like the idea of encoding all secrets in base64, and we could group by mime type in order to reduce duplication in the JSON. It would be more consistent and resolve this issue in a cleaner manner so I would prefer it. The reason I didn't include it in this PR is because it would definitely be a breaking change for users. If we are willing to deal with the change being breaking then I can definitely put it in.